pkg/providers/gcp/gcp.go

```diff
@@ -45,51 +51,98 @@ var (
 
 // GCP is the structure used as the provider for Google Cloud Platform
 type GCP struct {
-	client    *monitoring.QueryClient
-	projectID string
-	cache     *gcpCache
+	// GCP Clients
+	monitoring *monitoring.QueryClient
+	instances  *compute.InstancesClient
+
+	// Caching mechanism
+	cache *cache.Cache
+}
+
+// The GCP Resource representation
+type resource struct {
```

Comment:
why are you adding a new type? Shouldn't this use the `v1.Instance` struct

Reply:
Yeah I want to migrate it to the v1.Instance but this is how it was in the other file so I thought i'd do that on the next iteration as there was already so much change